lib/ex_fizz_buzz_enterprise/input_worker.ex

```diff
@@ -0,0 +1,35 @@
+defmodule ExFizzBuzzEnterprise.InputWorker do
+  use GenServer
+  require Logger
+  
+   @delay 500 # msec
+    
+  def start_link do
+    Logger.debug "Input Worker Starting"
+    
+    GenServer.start_link(__MODULE__, 
+      %ExFizzBuzzEnterprise.InputState{}, 
+      name: __MODULE__)
+  end
+  
+  def init(args) do
+    Logger.debug "Input Worker Initializing"
+    execute
+    {:ok, args}
```

Comment:
Better to use the timeout here and let `handle_info` play the part of loop iteration.